### DISCUSSION TITLE
Peer usage problem

### DISCUSSION THREAD

**Discussion by alexandre-embie** [tailwindlabs/tailwindcss#12004] [Help] [answered]
Hi, is there any reason why 
```jsx
<svg
className="absolute top-[4px] left-0"
xmlns="http://www.w3.org/2000/svg"
width="18"
height="18"
viewBox="0 0 18 18"
fill="none"
>
  <input
  type="radio"
  className="hidden-radio peer absolute inset-0 h-full w-full opacity-0"
  />
  <circle cx="9" cy="9" r="8" stroke="#211F20" />
  <circle className="invisible peer-checked:visible" cx="9" cy="9" r="6" fill="#211F20" />
</svg>
```

is not working but 
```jsx
<input
type="radio"
className="hidden-radio peer absolute inset-0 h-full w-full opacity-0"
/>
<svg
className="absolute top-[4px] left-0"
xmlns="http://www.w3.org/2000/svg"
width="18"
height="18"
viewBox="0 0 18 18"
fill="none"
>
    <circle cx="9" cy="9" r="8" stroke="#211F20" />
</svg>
<svg
className="invisible absolute top-[4px] left-0 peer-checked:visible"
xmlns="http://www.w3.org/2000/svg"
width="18"
height="18"
viewBox="0 0 18 18"
fill="none"
>
    <circle cx="9" cy="9" r="6" fill="#211F20" />
</svg>
```

is working ? I'm trying to avoid having 2 svgs here. Thanks for your help.

**Comment by wongjn** [comment, marked as answer]
The `<input>` inside the `<svg>` case does not work since the `<input>` element is no longer considered the "typical" HTML `<input>` element and does not have any of the characteristics of a radio form element.

To have the `<input>` outside the `<svg>` but have only one `<svg>` element while achieving what I am inferring to be your desired behavior, consider using [`group`](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state) in conjunction with `peer`-like behavior:

```html
<label>
  <input type="radio" class="… peer …"/>
  <svg class="… group">
    <!-- … -->
    <circle … class="invisible group-[.peer:checked~&]:visible" />
```

[Example on Tailwind Play](https://play.tailwindcss.com/ZnzdHukSD3).
